### ISSUE TITLE
Issue suspending with ath12k

### ISSUE THREAD

**Issue by dxlbnl** [pop-os/pop#3398]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
linux-firmware:
  Installed: 20240610.git1d67e66d-0pop1+system76~1718095302~22.04~81a2216
  Candidate: 20240610.git1d67e66d-0pop1+system76~1718095302~22.04~81a2216
  Version table:
 *** 20240610.git1d67e66d-0pop1+system76~1718095302~22.04~81a2216 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
     20220329.git681281e4-0ubuntu3.34 500
        500 http://apt.pop-os.org/ubuntu jammy-updates/main amd64 Packages
        500 http://apt.pop-os.org/ubuntu jammy-updates/main i386 Packages
     20220329.git681281e4-0ubuntu3.14 500
        500 http://apt.pop-os.org/ubuntu jammy-security/main amd64 Packages
        500 http://apt.pop-os.org/ubuntu jammy-security/main i386 Packages
     20220329.git681281e4-0ubuntu1 500
        500 http://apt.pop-os.org/ubuntu jammy/main amd64 Packages
        500 http://apt.pop-os.org/ubuntu jammy/main i386 Packages

**Issue/Bug Description:**
Running suspend fails due to a call in ath12k_pci failing (dpm_run_callback(): pci_pm_suspend+0x0/0x1b0 returns -95)

**Steps to reproduce (if you know):**
I'm running on a Thinkpad T14 Gen 5 AMD with a qualcomm NCM825 card.

**Expected behavior:**
Suspend suspending

**Other Notes:**
popos is prolly not the right place to post this. But I'm unsure where to file this bug.

**Comment by dxlbnl** [comment]
Additional logs (journalctl -f) during suspend: 


```
Oct 27 11:19:48 master ModemManager[1020]: <info>  [sleep-monitor-systemd] system is about to suspend
Oct 27 11:19:48 master sudo[10523]: pam_unix(sudo:session): session closed for user root
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.0821] manager: sleep: sleep requested (sleeping: no  enabled: yes)
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.0823] device (enp1s0f0): state change: unavailable -> unmanaged (reason 'sleeping', sys-iface-state: 'managed')
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.0907] device (p2p-dev-wlp2s0): state change: disconnected -> unmanaged (reason 'sleeping', sys-iface-state: 'managed')
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.0909] manager: NetworkManager state is now ASLEEP
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.0910] device (wlp2s0): state change: activated -> deactivating (reason 'sleeping', sys-iface-state: 'managed')
Oct 27 11:19:48 master gnome-shell[2362]: [popx11gestures@system76.com] Extension disabled
Oct 27 11:19:48 master gnome-shell[2362]: [popx11gestures@system76.com] Building entry point for GNOME Shell 42.9
Oct 27 11:19:48 master gnome-shell[2362]: [popx11gestures@system76.com] No tracker, skipping
Oct 27 11:19:48 master gnome-shell[2362]: [VERTICAL-OVERVIEW] resetting overrides
Oct 27 11:19:48 master gnome-shell[2362]: [VERTICAL-OVERVIEW] disabled
Oct 27 11:19:48 master systemd[1]: Starting Network Manager Script Dispatcher Service...
Oct 27 11:19:48 master systemd[1]: Started Network Manager Script Dispatcher Service.
Oct 27 11:19:48 master kernel: wlp2s0: deauthenticating from bc:30:d9:5b:ef:fb by local choice (Reason: 3=DEAUTH_LEAVING)
Oct 27 11:19:48 master gnome-shell[2362]: [popx11gestures@system76.com] Connection with Touchégg daemon lost: null
Oct 27 11:19:48 master gnome-shell[2362]: [popx11gestures@system76.com] Connection manually closed, not reconnecting to the daemon
Oct 27 11:19:48 master wpa_supplicant[990]: wlp2s0: CTRL-EVENT-DISCONNECTED bssid=bc:30:d9:5b:ef:fb reason=3 locally_generated=1
Oct 27 11:19:48 master wpa_supplicant[990]: wlp2s0: CTRL-EVENT-DSCP-POLICY clear_all
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.2324] device (wlp2s0): supplicant interface state: completed -> disconnected
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.2326] device (wlp2s0): state change: deactivating -> disconnected (reason 'sleeping', sys-iface-state: 'managed')
Oct 27 11:19:48 master avahi-daemon[951]: Withdrawing address record for fe80::8750:1351:ee44:5d76 on wlp2s0.
Oct 27 11:19:48 master avahi-daemon[951]: Leaving mDNS multicast group on interface wlp2s0.IPv6 with address fe80::8750:1351:ee44:5d76.
Oct 27 11:19:48 master avahi-daemon[951]: Interface wlp2s0.IPv6 no longer relevant for mDNS.
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.2416] dhcp4 (wlp2s0): canceled DHCP transaction
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.2417] dhcp4 (wlp2s0): activation: beginning transaction (timeout in 45 seconds)
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.2417] dhcp4 (wlp2s0): state changed no lease
Oct 27 11:19:48 master systemd-resolved[943]: wlp2s0: Bus client reset search domain list.
Oct 27 11:19:48 master avahi-daemon[951]: Withdrawing address record for 192.168.2.133 on wlp2s0.
Oct 27 11:19:48 master avahi-daemon[951]: Leaving mDNS multicast group on interface wlp2s0.IPv4 with address 192.168.2.133.
Oct 27 11:19:48 master systemd-resolved[943]: wlp2s0: Bus client set default route setting: no
Oct 27 11:19:48 master systemd-resolved[943]: wlp2s0: Bus client reset DNS server list.
Oct 27 11:19:48 master avahi-daemon[951]: Interface wlp2s0.IPv4 no longer relevant for mDNS.
Oct 27 11:19:48 master NetworkManager[10326]: <info>  [1730024388.2825] device (wlp2s0): state change: disconnected -> unmanaged (reason 'sleeping', sys-iface-state: 'managed')
Oct 27 11:19:48 master chronyd[1087]: Source 3.220.42.39 offline
Oct 27 11:19:48 master chronyd[1087]: Source 45.83.205.227 offline
Oct 27 11:19:48 master chronyd[1087]: Source 95.85.21.89 offline
Oct 27 11:19:48 master chronyd[1087]: Source 185.125.190.58 offline
Oct 27 11:19:48 master chronyd[1087]: Source 52.10.183.132 offline
Oct 27 11:19:48 master chronyd[1087]: Source 213.109.127.82 offline
Oct 27 11:19:48 master chronyd[1087]: Source 45.138.55.60 offline
Oct 27 11:19:48 master chronyd[1087]: Source 185.125.190.57 offline
Oct 27 11:19:48 master chronyd[1087]: Source 3.134.129.152 offline
Oct 27 11:19:48 master chronyd[1087]: Source 18.228.202.30 offline
Oct 27 11:19:48 master chronyd[1087]: Source 185.125.190.56 offline
Oct 27 11:19:48 master chronyd[1087]: Source 91.189.91.157 offline
Oct 27 11:19:48 master chronyd[1087]: Can't synchronise: no selectable sources
Oct 27 11:19:48 master chronyd[1087]: Source 15.237.97.214 offline
Oct 27 11:19:48 master wpa_supplicant[990]: p2p-dev-wlp2s0: CTRL-EVENT-DSCP-POLICY clear_all
Oct 27 11:19:48 master wpa_supplicant[990]: p2p-dev-wlp2s0: CTRL-EVENT-DSCP-POLICY clear_all
Oct 27 11:19:48 master wpa_supplicant[990]: nl80211: deinit ifname=p2p-dev-wlp2s0 disabled_11b_rates=0
Oct 27 11:19:48 master wpa_supplicant[990]: wlp2s0: CTRL-EVENT-DSCP-POLICY clear_all
Oct 27 11:19:48 master wpa_supplicant[990]: wlp2s0: CTRL-EVENT-DSCP-POLICY clear_all
Oct 27 11:19:48 master wpa_supplicant[990]: nl80211: deinit ifname=wlp2s0 disabled_11b_rates=0
Oct 27 11:19:48 master gnome-shell[2362]: An active wireless connection, in infrastructure mode, involves no access point?
Oct 27 11:19:51 master systemd[1]: Reached target Sleep.
Oct 27 11:19:51 master systemd[1]: Starting System Suspend...
Oct 27 11:19:51 master systemd[1]: Starting Load/Save RF Kill Switch Status...
Oct 27 11:19:51 master systemd[1]: Started Load/Save RF Kill Switch Status.
Oct 27 11:19:51 master kernel: usb 1-3.1: USB disconnect, device number 4
Oct 27 11:19:51 master kernel: Bluetooth: hci0: Opcode 0x0c1a failed: -19
Oct 27 11:19:51 master kernel: Bluetooth: hci0: Error when powering off device on rfkill (-19)
Oct 27 11:19:51 master rfkill[10576]: block set for type bluetooth
Oct 27 11:19:51 master systemd-sleep[10570]: Entering sleep state 'suspend'...
Oct 27 11:19:51 master kernel: PM: suspend entry (s2idle)
Oct 27 11:19:51 master systemd[2172]: Stopped target Bluetooth.
Oct 27 11:19:51 master systemd[1]: Stopped target Bluetooth Support.
Oct 27 11:19:51 master /usr/libexec/gdm-x-session[2214]: (II) event14 - Logitech MX Master 3S: device removed
Oct 27 11:19:51 master /usr/libexec/gdm-x-session[2214]: (II) config/udev: removing device Logitech MX Master 3S
Oct 27 11:19:51 master /usr/libexec/gdm-x-session[2214]: (**) Option "fd" "62"
Oct 27 11:19:51 master /usr/libexec/gdm-x-session[2214]: (II) UnloadModule: "libinput"
Oct 27 11:19:51 master /usr/libexec/gdm-x-session[2214]: (II) systemd-logind: releasing fd for 13:78
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/ldac
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/aptx_hd
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aptx_hd
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/aptx
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aptx
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/aac
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aac
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/sbc
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/sbc
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/sbc_xq
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/sbc_xq
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aptx_ll_1
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aptx_ll_0
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aptx_ll_duplex_1
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/aptx_ll_duplex_0
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/faststream
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/faststream_duplex
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/opus_05
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/opus_05
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSink/opus_05_duplex
Oct 27 11:19:51 master bluetoothd[952]: Endpoint unregistered: sender=:1.76 path=/MediaEndpoint/A2DPSource/opus_05_duplex
Oct 27 11:19:51 master kernel: Filesystems sync: 0.031 seconds
Oct 27 11:19:51 master upowerd[1046]: g_source_remove: assertion 'tag > 0' failed
Oct 27 11:19:52 master kernel: Freezing user space processes
Oct 27 11:19:52 master kernel: Freezing user space processes completed (elapsed 0.007 seconds)
Oct 27 11:19:52 master kernel: OOM killer disabled.
Oct 27 11:19:52 master kernel: Freezing remaining freezable tasks
Oct 27 11:19:52 master kernel: Freezing remaining freezable tasks completed (elapsed 0.075 seconds)
Oct 27 11:19:52 master kernel: printk: Suspending console(s) (use no_console_suspend to debug)
Oct 27 11:19:52 master kernel: ath12k_pci 0000:02:00.0: failed to suspend core: -95
Oct 27 11:19:52 master kernel: ath12k_pci 0000:02:00.0: PM: pci_pm_suspend(): ath12k_pci_pm_suspend+0x0/0x60 [ath12k] returns -95
Oct 27 11:19:52 master kernel: ath12k_pci 0000:02:00.0: PM: dpm_run_callback(): pci_pm_suspend+0x0/0x1b0 returns -95
Oct 27 11:19:52 master kernel: ath12k_pci 0000:02:00.0: PM: failed to suspend async: error -95
Oct 27 11:19:52 master kernel: PM: Some devices failed to suspend, or early wake event detected
Oct 27 11:19:52 master kernel: [drm] PCIE GART of 512M enabled (table at 0x000000803FD00000).
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: SMU is resuming...
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: SMU is resumed successfully!
Oct 27 11:19:52 master kernel: nvme nvme0: D3 entry latency set to 10 seconds
Oct 27 11:19:52 master kernel: nvme nvme0: 16/0/0 default/read/poll queues
Oct 27 11:19:52 master kernel: [drm] VCN decode and encode initialized successfully(under DPG Mode).
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: [drm:jpeg_v4_0_hw_init [amdgpu]] JPEG decode initialized successfully.
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring gfx_0.0.0 uses VM inv eng 0 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 6 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 7 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 8 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 9 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 10 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 11 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring sdma0 uses VM inv eng 12 on hub 0
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring vcn_unified_0 uses VM inv eng 0 on hub 8
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring jpeg_dec uses VM inv eng 1 on hub 8
Oct 27 11:19:52 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring mes_kiq_3.1.0 uses VM inv eng 13 on hub 0
Oct 27 11:19:52 master kernel: [drm] ring gfx_32778.1.1 was added
Oct 27 11:19:52 master kernel: [drm] ring compute_32778.2.2 was added
Oct 27 11:19:52 master kernel: [drm] ring sdma_32778.3.3 was added
Oct 27 11:19:52 master kernel: [drm] ring gfx_32778.1.1 ib test pass
Oct 27 11:19:52 master kernel: [drm] ring compute_32778.2.2 ib test pass
Oct 27 11:19:52 master kernel: [drm] ring sdma_32778.3.3 ib test pass
Oct 27 11:19:52 master kernel: OOM killer enabled.
Oct 27 11:19:52 master kernel: Restarting tasks ... done.
Oct 27 11:19:52 master kernel: random: crng reseeded on system resumption
Oct 27 11:19:52 master kernel: PM: suspend exit
Oct 27 11:19:52 master kernel: PM: suspend entry (s2idle)
Oct 27 11:19:54 master kernel: Filesystems sync: 0.022 seconds
Oct 27 11:19:54 master kernel: Freezing user space processes
Oct 27 11:19:54 master kernel: Freezing user space processes completed (elapsed 0.236 seconds)
Oct 27 11:19:54 master kernel: OOM killer disabled.
Oct 27 11:19:54 master kernel: Freezing remaining freezable tasks
Oct 27 11:19:54 master kernel: Freezing remaining freezable tasks completed (elapsed 0.001 seconds)
Oct 27 11:19:54 master kernel: printk: Suspending console(s) (use no_console_suspend to debug)
Oct 27 11:19:54 master kernel: ath12k_pci 0000:02:00.0: failed to suspend core: -95
Oct 27 11:19:54 master kernel: ath12k_pci 0000:02:00.0: PM: pci_pm_suspend(): ath12k_pci_pm_suspend+0x0/0x60 [ath12k] returns -95
Oct 27 11:19:54 master kernel: ath12k_pci 0000:02:00.0: PM: dpm_run_callback(): pci_pm_suspend+0x0/0x1b0 returns -95
Oct 27 11:19:54 master kernel: ath12k_pci 0000:02:00.0: PM: failed to suspend async: error -95
Oct 27 11:19:54 master kernel: PM: Some devices failed to suspend, or early wake event detected
Oct 27 11:19:54 master kernel: [drm] PCIE GART of 512M enabled (table at 0x000000803FD00000).
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: SMU is resuming...
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: SMU is resumed successfully!
Oct 27 11:19:54 master kernel: nvme nvme0: D3 entry latency set to 10 seconds
Oct 27 11:19:54 master kernel: nvme nvme0: 16/0/0 default/read/poll queues
Oct 27 11:19:54 master kernel: [drm] VCN decode and encode initialized successfully(under DPG Mode).
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: [drm:jpeg_v4_0_hw_init [amdgpu]] JPEG decode initialized successfully.
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring gfx_0.0.0 uses VM inv eng 0 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 6 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 7 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 8 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 9 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 10 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 11 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring sdma0 uses VM inv eng 12 on hub 0
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring vcn_unified_0 uses VM inv eng 0 on hub 8
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring jpeg_dec uses VM inv eng 1 on hub 8
Oct 27 11:19:54 master kernel: amdgpu 0000:c4:00.0: amdgpu: ring mes_kiq_3.1.0 uses VM inv eng 13 on hub 0
Oct 27 11:19:54 master kernel: [drm] ring gfx_32778.1.1 was added
Oct 27 11:19:54 master kernel: [drm] ring compute_32778.2.2 was added
Oct 27 11:19:54 master kernel: [drm] ring sdma_32778.3.3 was added
Oct 27 11:19:54 master kernel: [drm] ring gfx_32778.1.1 ib test pass
Oct 27 11:19:54 master kernel: [drm] ring compute_32778.2.2 ib test pass
Oct 27 11:19:54 master kernel: [drm] ring sdma_32778.3.3 ib test pass
Oct 27 11:19:54 master kernel: OOM killer enabled.
Oct 27 11:19:54 master kernel: Restarting tasks ... done.
Oct 27 11:19:54 master kernel: random: crng reseeded on system resumption
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): EDID vendor "SDC", prod id 16799
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): Printing DDC gathered Modelines:
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): Modeline "2880x1800"x0.0  652.26  2880 2912 2920 2980  1800 1808 1816 1824 +hsync -vsync (218.9 kHz eP)
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): Modeline "2880x1800"x0.0  652.26  2880 2888 2920 2980  1800 1808 1816 3648 +hsync -vsync (218.9 kHz e)
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): EDID vendor "SDC", prod id 16799
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): Printing DDC gathered Modelines:
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): Modeline "2880x1800"x0.0  652.26  2880 2912 2920 2980  1800 1808 1816 1824 +hsync -vsync (218.9 kHz eP)
Oct 27 11:19:54 master /usr/libexec/gdm-x-session[2214]: (II) AMDGPU(0): Modeline "2880x1800"x0.0  652.26  2880 2888 2920 2980  1800 1808 1816 3648 +hsync -vsync (218.9 kHz e)
Oct 27 11:19:54 master systemd-sleep[10570]: Failed to put system to sleep. System resumed again: Operation not supported
Oct 27 11:19:54 master kernel: PM: suspend exit
Oct 27 11:19:54 master gnome-shell[2362]: JS ERROR: Failed to initialize fprintd service: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
                                          asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:114:23
Oct 27 11:19:55 master rfkill[10648]: unblock set for type bluetooth
Oct 27 11:19:55 master systemd[1]: systemd-suspend.service: Main process exited, code=exited, status=1/FAILURE
Oct 27 11:19:55 master systemd[1]: systemd-suspend.service: Failed with result 'exit-code'.
Oct 27 11:19:55 master systemd[1]: Failed to start System Suspend.
Oct 27 11:19:55 master systemd[1]: Dependency failed for Suspend.
Oct 27 11:19:55 master systemd[1]: suspend.target: Job suspend.target/start failed with result 'dependency'.
Oct 27 11:19:55 master systemd-logind[975]: Operation 'sleep' finished.
Oct 27 11:19:55 master systemd[1]: Stopped target Sleep.
Oct 27 11:19:55 master ModemManager[1020]: <info>  [sleep-monitor-systemd] system is resuming

```

**Comment by dxlbnl** [comment]
Appearantly its fixed in https://gitlab.com/kernel-firmware/linux-firmware/-/tags/20241017 but pop-os is out of date a bunch of releases

**Comment by jwilczek** [comment]
I'm also seeing this issue. Is there any workaround?

**Comment by dxlbnl** [comment]
I'm unsure if it's possible to update linux-firmware to the latest version (from the repo). Ubuntu and debian also don't have this package updated yet.

**Comment by dxlbnl** [comment]
I've installed fedora, and it's running much smoother. Though I keep having issues with my bluetooth headset. Its a compromise I'm okay with.

**Comment by davidphysdavalos** [comment]
With the help of chatGPT, the **WORKAROUND** I am using is just to unload the module before suspending and reload it in resume: 

**Step 1:** add or edit: `sudo nano /etc/systemd/sleep.conf.d/wifi.conf` with the contents:

`[Sleep]
SuspendState=mem
AllowSuspend=yes
AllowHibernation=no
AllowSuspendThenHibernate=no`

**Step 2:** add or edit: `sudo nano /etc/modprobe.d/ath12k_suspend.conf` with the contents:

`options ath12k_pci disable_aspm=1`


**Step 3:** Add or edit: ` sudo nano /lib/systemd/system-sleep/ath12k-unload.sh` with:

`#!/bin/bash
case $1/$2 in
  pre/*)
    modprobe -r ath12k
    ;;
  post/*)
    modprobe ath12k
    ;;
esac`

**Step 4:** `sudo chmod +x /lib/systemd/system-sleep/ath12k-unload.sh`
